### PR TITLE
NOISSUE - add subtopic support on http adapter

### PR DESCRIPTION
**Pull request by beres**
### What does this do?
The patch add subtopic support for http-adapter

### CODE REVIEW

**Comment by beres** [comment]
The same approach can be used for ws/coap adapters, so the regexp could be exported in a more generic place, topics.go? What do you think?

File: http/api/transport.go [hunk 1]
```diff
@@ -27,8 +29,9 @@ import (
 const protocol = "http"
 
 var (
-	errMalformedData = errors.New("malformed request data")
-	auth             mainflux.ThingsServiceClient
+	errMalformedData  = errors.New("malformed request data")
+	auth              mainflux.ThingsServiceClient
+	channelPartRegExp = regexp.MustCompile(`^/channels/(?P<channelId>[\w\-]+)/messages(?P<subTopics>/.+[^/])*$`)
```
Comment (drasko) [inline http/api/transport.go]: Use `chanID` instead of `channelId` to stay Go-idiomatic.

File: http/api/transport.go [hunk 2]
```diff
@@ -81,23 +89,17 @@ func decodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
 	return msg, nil
 }
 
-func authorize(r *http.Request) (string, error) {
+func authorize(r *http.Request, channelId string) (string, error) {
```
Comment (drasko) [inline http/api/transport.go]: `chanID`

File: http/api/transport.go [hunk 3]
```diff
@@ -81,23 +89,17 @@ func decodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
...
-	chanID := bone.GetValue(r, "id")
-	if chanID == "" {
-		return "", errMalformedData
-	}
-
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
 
-	id, err := auth.CanAccess(ctx, &mainflux.AccessReq{Token: apiKey, ChanID: chanID})
+	id, err := auth.CanAccess(ctx, &mainflux.AccessReq{Token: apiKey, ChanID: channelId})
```
Comment (drasko) [inline http/api/transport.go]: `chanID`

**Review by drasko — CHANGES_REQUESTED** [review]

**Comment by drasko** [comment]
Related to https://github.com/mainflux/mainflux/issues/596 - @beres can you take this one? It's not trivial :)

**Comment by nmarcetic** [comment]
Duplicate #642

**Pull request closed without merging**